src/player.js

```diff
@@ -286,6 +299,33 @@ class Player extends FakeEventTarget {
 
   //  <editor-fold desc="Playback Interface">
   /**
+   * The player readiness
+   * @public
+   * @returns {Promise<*>} - The ready promise
+   */
+  ready(): Promise<*> {
+    return this._readyPromise ? this._readyPromise : Promise.resolve();
```

Comment:
I don't understand. the ready promise must be null on change media, else it will be resolved
before the new media will loaded. and on change media we don't create a new player and the constructor not invoked.